buildkite/resource_cluster.go

```diff
@@ -0,0 +1,216 @@
+		return
+	}
+
+	_, err := getCluster(c.client.genqlient, c.client.organization, state.UUID.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read Cluster",
+			fmt.Sprintf("Unable to read Cluster: %s", err.Error()),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (c *clusterResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state, plan clusterResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	id := state.ID.ValueString()
+
+	_, err := updateCluster(
+		c.client.genqlient,
+		c.client.organizationId,
+		id,
+		plan.Name.ValueString(),
+		plan.Description.ValueString(),
+		plan.Emoji.ValueString(),
+		plan.Color.ValueString(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to update Cluster",
+			fmt.Sprintf("Unable to update Cluster: %s", err.Error()),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (c *clusterResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state clusterResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := deleteCluster(c.client.genqlient, c.client.organizationId, state.ID.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to delete Cluster",
+			fmt.Sprintf("Unable to delete Cluster: %s", err.Error()),
+		)
+		return
+	}
+}
+
+func (c *clusterResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	uuid := req.ID
+
+	if uuid == "" {
+		resp.Diagnostics.AddError(
+			"Unable to import Cluster",
+			"Unable to import Cluster, no UUID was provided",
+		)
+		return
+	}
+
+	cluster, err := getCluster(c.client.genqlient, c.client.organization, uuid)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to import Cluster",
+			fmt.Sprintf("Unable to import Cluster: %s", err.Error()),
+		)
+		return
+	}
+
+	id := cluster.Organization.Cluster.Id
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), types.StringValue(id))...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("uuid"), uuid)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("name"), types.StringValue(cluster.Organization.Cluster.Name))...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("description"), types.StringValue(cluster.Organization.Cluster.Description))...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("emoji"), types.StringValue(cluster.Organization.Cluster.Emoji))...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("color"), types.StringValue(cluster.Organization.Cluster.Color))...)
```

Comment:
These should be moved to the Read function. That function will be called after ImportState anyway and Read should set these values. ImportState is responsible for setting the minimal fields for Read to work

Reply:
Also got a hunch that this is why the import test is failing - similar behaviour with queues